File: leetcode1177/bitPrefix.cpp

// class Solution 
// {
// public:
//     vector<bool> canMakePaliQueries(string s, vector<vector<int>>& queries) 
//     {
//         const int n=s.length(),nq=queries.size();
//         vector<bool>ans(nq);

//         vector<int>prefix(n+1);

//         int i;
//         for(i=0;i<n;++i)
//             prefix[i+1]=prefix[i],
//                 prefix[i+1]^=1 << s[i]-'a';
        
//         int j,len,nOdd;
//         //array<int,26>tmp;
//         for(i=0;i<nq;++i)
//         {
//             nOdd=__builtin_popcount(prefix[queries[i][1]+1]^prefix[queries[i][0]]);
//             //总字母个数        奇数个字母的个数
//             //len=queries[i][1]-queries[i][0]+1;
//             //nEven=len - nOdd
//             ans[i]=nOdd>>1 <= queries[i][2];
//         }

//         return ans;
//     }
// };

File: leetcode1177/bitPrefix.go

// func canMakePaliQueries(s string, queries [][]int) []bool {

// 	n := len(s)

// 	nq := len(queries)

// 	ans := make([]bool, nq)

// 	prefix := make([]int, n+1)

// 	for i := 0; i < n; i++ {

// 		prefix[i+1] = prefix[i] ^ (1 << (s[i] - 'a'))

// 	}

// 	canMakePalindrome := func(start, end, k int) bool {

// 		nOdd := bits.OnesCount(uint(prefix[end+1] ^ prefix[start]))

// 		return nOdd>>1 <= k

// 	}

// 	for i := 0; i < nq; i++ {

// 		ans[i] = canMakePalindrome(queries[i][0], queries[i][1], queries[i][2])

// 	}

// 	return ans

// }